lib/Dialect/RTL/Ops.cpp

```diff
@@ -470,7 +479,13 @@ void OrOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
         rewriter.replaceOpWithNewOp<OrOp>(op, op.getType(), inputs.drop_back());
         return success();
       }
-      /// TODO: or(..., x, x) -> or(..., x) -- idempotent
+
+      // or(..., x, x) -> or(..., x) -- idempotent
+      if (inputs[size - 1] == inputs[size - 2]) {
+        rewriter.replaceOpWithNewOp<OrOp>(op, op.getType(), inputs.drop_back());
```

Comment:
Please add the corresponding cases for 'and' and 'xor', the former can just drop the operand like or does, the later drop both operands (and produces a zero if none are left)

Reply:
Sure. I had a question, since I'm running in circles a bit trying to find this out. I want to declare the zero value:

```
   auto zero = rewriter.create<ConstantOp>(
              op.getLoc(), APInt(/*numBits=*/11, /*val=*/0)); 
```
However, obviously we don't want to just hardcode the number of bits. Given that the inputs are non-constants, how does one get the number of bits?

Reply:
It is a bit annoying, because you need to dig the width out of the integer type.  I'll add a helper for this, one moment.

Reply:
Ok, with aee1272 you should be able to do something like this:

uto zero = rewriter.create<ConstantOp>(
              op.getLoc(), 0, op.getType().cast<IntegerType>());

Reply:
I believe `aee1272` is giving some build errors. Let me know if you want the exact errors and I'll paste them here.

Reply:
taking a look now, I just saw the builder is failing.  It works for me locally, feel free to revert if it is blockign your progress but I'm taking a look now

Reply:
Fixed in 92b4d86.  somehow I didn't commit all the changes?? Sorry about that!

Reply:
It works, thanks for the help :)